packages/datagateway-table/src/breadcrumbs.component.tsx

```diff
@@ -0,0 +1,538 @@
+      // Get the entity and the data stored on the entity.
+      let entity = this.currentPathnames[index];
+
+      const link = `/${this.currentPathnames.slice(0, index + 3).join('/')}`;
+
+      // Check we are not the end of the pathnames array.
+      if (index < pathLength - 1) {
+        // If the index does not already exist,
+        // create the entry for the index.
+        if (!updatedState.currentHierarchy[hierarchyCount]) {
+          // Insert the new breadcrumb information into the array.
+          updatedState.currentHierarchy.splice(hierarchyCount, 1, {
+            id: '',
+            pathName: entity,
+            displayName: 'N/A',
+            url: link,
+          });
+        }
+
+        // Get the information held in the state for the current path name.
+        const entityInfo = updatedState.currentHierarchy[hierarchyCount];
+
+        // Get the entity Id in the path to compare with the saved one in our state.
+        const entityId = this.currentPathnames[index + 1];
+
+        // Check if an entity id is present or if the id has changed since the last update to the state.
+        if (entityInfo.id.length === 0 || entityInfo.id !== entityId) {
+          // In general the API endpoint will be our entity name and
+          // the entity field we want is the NAME of the entity.
+          let apiEntity = entity;
+          let requestEntityField = 'NAME';
+
+          // There are some exceptions to this when handling the DIAMOND/DLS
+          // depending on if 'proposal' has been found in the current path.
+          if (this.currentPathnames.includes('proposal')) {
+            // If the entity is current proposal, then we will not make an API request,
+            // as we need the investigation ID to retrieve the TITLE.
+            if (entity === 'proposal') {
+              apiEntity = 'investigation';
+              requestEntityField = 'TITLE';
+            } else if (entity === 'investigation') {
+              // Otherwise, we can proceed and get the VISIT_ID for the investigation entity.
+              requestEntityField = 'VISIT_ID';
+            }
+          } else {
+            // Anything else (including ISIS), we request the TITLE for the investigation entity.
+            if (entity === 'investigation') requestEntityField = 'TITLE';
+          }
+
+          // Create the entity url to request the name, this is pluralised to get the API endpoint.
+          let requestEntityUrl;
+          if (entity !== 'proposal') {
+            requestEntityUrl = `${apiEntity}s`.toLowerCase() + `/${entityId}`;
+          } else {
+            // If we are searching for proposal, we know that there is no investigation
+            // information in the current path. We will need to query and select one investigation
+            // from all investigations with the entity id (which is the proposal/investigation name).
+            requestEntityUrl =
+              `${apiEntity}s`.toLowerCase() +
```

Comment:
internationalisation

Reply:
This is an api call and thus doesn't need to be internationalised (except maybe to pluralise in english properly?)